r/tests/testthat/test-dplyr.R

```diff
@@ -945,3 +945,63 @@ test_that("abs()", {
     df
   )
 })
+
+test_that("if_else and ifelse", {
+  df <- tibble(x = c(-127, -10, -1, -0 , 0, 1, 10, 127, NA))
```

Comment:
Is there a reason we only test with type numeric? Do we trust that all types are faithfully tested in C++?

Reply:
Good point. I'll look and see if they are, if not I'll add some.

Reply:
Here are (some of) the c++ tests, which look pretty comprehensive for the types that are currently supported: https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/kernels/scalar_if_else_test.cc#L49-L94